src/App.tsx

```diff
@@ -2,10 +2,46 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.target?.files?.[0]) {
+      const file = e.target.files[0]
+
+      const reader = new FileReader()
+      reader.onload = () => {
+        const xml = reader.result?.toString()
+
+        if (xml) {
+          xml2js.parseString(xml, { trim: true }, async function(err, result) {
+            //todo: display error nicely in UI
+            if (err) {
+              console.log(err)
```

Comment:
If it does error out, you probably don't want to try parsing the results (or at least don't want to waste the effort, even if you make the parser robust enough), so you might want to `return` in this block as well (in addition to logging it to the UI nicely - for which a `const [error, setError] = React.useState<String>('')` is probably warranted.)

Reply:
changed to xml2js.parseStringPromise and handle the error that way